draft-west-cookie-incrementalism.md

```diff
@@ -224,6 +233,48 @@ attribute (Section 4.1.2.5 of {{RFC6265bis}}) by altering the storage model defi
 This is conceptually similar to the requirements put into place for the `__Secure-` prefix (Section
 4.1.3.1 of {{RFC6265bis}}).
 
+## Schemeful Same-Site {#schemeful-samesite}
+
+By using the scheme, as well as the registrable domain, schemeful same-site can
+help to protect https origins against a network attacker that is impersonating
+an http origin with the same registrable domain, thereby further increasing
+SameSite's CSRF protections. To do so we need to modify a number of things:
+
+First change the definition of "site for cookies" from a registrable domain to
+an origin. In the places where a we return an empty string for a non-existent
+"site for cookies" we should instead return an origin set to a freshly
+generated globally unique identifier.
+
+Then replace the same-site calculation algorithm with the following:
+
+~~~
+Two origins, A and B, are considered same-site if the following algorithm returns true:
+1.  If A and B are both scheme/host/port triples then
```

Comment:
1.  Nit: Newline above.
2.  It's probably best to shift towards HTML's terminology here of "tuple origin" (vs "opaque origin").

Reply:
1. Done
2. I was trying to stick close to the IETF terminology given that this will, ideally, become part of an IETF doc in order to keep terms internally consistent. That being said, if you'd still like me to change over to the HTML version I can do that. It'll make the wording less awkward as well.

Reply:
What about RFC 6454 is still relevant to web browsers?

Reply:
The algorithms are all outdated, as are the UA requirements. Some of the non-normative text is valuable. I think all the important parts are covered in URL and HTML. If @mnot, et al. are happy to consider the RFC obsoleted by those specifications, I'm happy to link to them. Based on my vague recollection of some discussions of URLs, my intuition is that there'd be some dispute about that.